bin/ruby-build

```diff
@@ -1063,10 +1063,18 @@ build_package_mac_openssl() {
 
   build_package_standard "$@"
 
-  # Extract root certs from the system keychain in .pem format and rehash.
   local pem_file="$OPENSSLDIR/cert.pem"
-  security find-certificate -a -p /Library/Keychains/System.keychain > "$pem_file"
-  security find-certificate -a -p /System/Library/Keychains/SystemRootCertificates.keychain >> "$pem_file"
+  if is_mac; then
+    # Extract root certs from the system keychain in .pem format and rehash.
+    security find-certificate -a -p /Library/Keychains/System.keychain > "$pem_file"
+    security find-certificate -a -p /System/Library/Keychains/SystemRootCertificates.keychain >> "$pem_file"
+  else
+    # See https://github.com/rubygems/rubygems/issues/2415#issuecomment-509806259
+    # Debian-based logic:
+    rm -rf "$OPENSSLDIR/certs" "$pem_file"
+    ln -s /etc/ssl/certs "$OPENSSLDIR"
+    ln -s /etc/ssl/certs/ca-certificates.crt "$pem_file"
+  fi
```

Comment:
This logic needs more tweaks, it needs to support other OS, and have some kind of fallback.
We probably want to rename `build_package_mac_openssl` to just `build_package_openssl` or so.
In general I think it would be better to have 2 PRs:
* One updating OpenSSL and no other changes.
* One building OpenSSL on Linux for Ruby <= 2.3 (since it's hard to get OpenSSL 1.0.2 from Linux distributions).